### ISSUE TITLE
Specify Zones

### ISSUE THREAD

**Issue by rubydotexe** [libdns/googleclouddns#5]
Hi! I have multiple zones in my Cloud DNS, and I was wondering if there could be a variable that allows me to specify which domain zone(s) I am using when calling the Google Cloud API. I am using Caddy with the Google Cloud DNS plugin, and abstractly, being unable to specify between zones means I get this error: `unable to find Google managaged zone for domain xyz.` 

An adjustment here would also mean that the github.com/caddy-dns/googleclouddns would need to be adjusted too, but I would make a new issue there if my request is approved.

**Comment by brockwood** [comment]
Hello @rubydotexe and apologies for the delay in replying.

Could you please give an example of when this occurs? Currently, when the googleclouddns client starts up, it will list all the zones for the specified project and then match up the DNS zone name passed in from Caddy with the zone in Google Cloud. The only catch is that the zone has to be a public zone; if it is private, the client will ignore it.

Any additional information you could provide would be appreciated, thanks!

**Comment by brockwood** [comment]
Hey @rubydotexe. thank you for the additional context. I've been trying to reproduce your issue and I've not had any luck. I have a GCP project with a public zone `wxgl.us.` and a wildcard entry pointing at `127.0.0.1` for testing purposes:

![image](https://github.com/libdns/googleclouddns/assets/739721/2947e976-3d9c-4e0e-8397-a2804c001017)

I am using the JSON file from a service account with DNS admin rights in the project and I have the following environment variables defined:

- `GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json`
- `GCP_PROJECT=project-name`

I then am using the following `Caddyfile`:

```caddyfile
*.wxgl.us {
	bind 127.0.0.1
	tls <email address> {
		dns googleclouddns {
			gcp_project {$GCP_PROJECT}
		}
	}
	respond "Hello there! I'm a Caddy!"
}
```

I'm able to generate a cert in this configuration:

```
2023/12/20 17:42:26.419 INFO    tls.obtain      acquiring lock  {"identifier": "*.wxgl.us"}
2023/12/20 17:42:26.425 INFO    tls     cleaning storage unit   {"storage": "FileStorage:/home/caddy/.local/share/caddy"}
2023/12/20 17:42:26.426 INFO    tls     finished cleaning storage units
2023/12/20 17:42:26.430 INFO    tls.obtain      lock acquired   {"identifier": "*.wxgl.us"}
2023/12/20 17:42:26.431 INFO    tls.obtain      obtaining certificate   {"identifier": "*.wxgl.us"}
2023/12/20 17:42:26.720 INFO    tls.issuance.acme       waiting on internal rate limiter        {"identifiers": ["*.wxgl.us"], "ca": "https://acme-v02.api.letsencrypt.org/directory", "account": "user@home.com"}
2023/12/20 17:42:26.720 INFO    tls.issuance.acme       done waiting on internal rate limiter   {"identifiers": ["*.wxgl.us"], "ca": "https://acme-v02.api.letsencrypt.org/directory", "account": "user@@home.com"}
2023/12/20 17:42:26.944 INFO    tls.issuance.acme.acme_client   trying to solve challenge       {"identifier": "*.wxgl.us", "challenge_type": "dns-01", "ca": "https://acme-v02.api.letsencrypt.org/directory"}
2023/12/20 17:42:30.499 INFO    tls.issuance.acme.acme_client   authorization finalized {"identifier": "*.wxgl.us", "authz_status": "valid"}
2023/12/20 17:42:30.499 INFO    tls.issuance.acme.acme_client   validations succeeded; finalizing order {"order": "https://acme-v02.api.letsencrypt.org/acme/order/<order number>/<redacted>"}
2023/12/20 17:42:31.391 INFO    tls.issuance.acme.acme_client   successfully downloaded available certificate chains    {"count": 2, "first_url": "https://acme-v02.api.letsencrypt.org/acme/cert/<redacted>"}
2023/12/20 17:42:31.392 INFO    tls.obtain      certificate obtained successfully       {"identifier": "*.wxgl.us"}
2023/12/20 17:42:31.392 INFO    tls.obtain      releasing lock  {"identifier": "*.wxgl.us"}
```

I'm hoping you can share how this setup differs with your own to see if we can figure out what's going on. Thanks!
